tmp.yml

```diff
@@ -0,0 +1,66 @@
+# Also, TechPreview featureset is required here
+# https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-cli_nodes-cluster-enabling
+
+# The nodeSelector is evaluated during pod scheduling only, that means that if you change the node labels after the pod was created it won't move.
+# For your test case this means that you have to label the node before changing the network.config settings
+
+# If no match node, the the pod will be in pending status, right ?
+# Pending means it is waiting for an available node
```

Comment:
That's correct. If there are no nodes available to host he network-check-source deployment the pod will be stuck in pending until such node becomes available.